### PR TITLE
LPS-69102 Move the General file uploads settings UI from Server Administration to System Settings

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-69102](https://issues.liferay.com/browse/LPS-69102).

Hey @JorgeFerrer, I wanted you to look at this pull since it has a lot of unique changes.  @pei-jung is also peer reviewing.

/cc @pei-jung

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:retest

**Comment by JorgeFerrer** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/foundation/file-uploads/file-uploads-configuration/src/main/java/com/liferay/file/uploads/configuration/listener/GeneralFileUploadsConfigurationModelListener.java [hunk 1]
```diff
@@ -14,11 +14,108 @@
...
+
+	@Override
+	public void onAfterSave(String pid, Dictionary<String, Object> properties)
+		throws ConfigurationModelListenerException {
+
+		_updateSettings(properties);
+	}
+
+	private void _updateSettings(Dictionary<String, Object> properties)
+		throws ConfigurationModelListenerException {
```
Comment (JorgeFerrer) [inline modules/apps/foundation/file-uploads/file-uploads-configuration/src/main/java/com/liferay/file/uploads/configuration/listener/GeneralFileUploadsConfigurationModelListener.java]: Hey Drew,

Can you check with Brian if he's ok with this approach as a temporary solution?
It's the best solution I can think of, but maybe he prefers a different approach.

File: modules/apps/foundation/file-uploads/file-uploads-configuration/src/main/java/com/liferay/file/uploads/configuration/listener/GeneralFileUploadsConfigurationModelListener.java [hunk 2]
```diff
@@ -14,11 +14,108 @@
...
+				String.valueOf(
+					generalFileUploadsConfiguration.
+						uploadServletRequestImplMaxSize()));
+
+			String tempDir =
+				generalFileUploadsConfiguration.
+					uploadServletRequestImplTempDir();
+
+			if (Validator.isNull(tempDir)) {
+				tempDir = SystemProperties.get(SystemProperties.TMP_DIR);
```
Comment (JorgeFerrer) [inline modules/apps/foundation/file-uploads/file-uploads-configuration/src/main/java/com/liferay/file/uploads/configuration/listener/GeneralFileUploadsConfigurationModelListener.java]: Why is this needed?

File: modules/apps/foundation/file-uploads/file-uploads-configuration/src/main/java/com/liferay/file/uploads/configuration/verify/GeneralFileUploadsVerifyProcess.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+/**
+ * @author Drew Brokke
+ */
+@Component(
+	immediate = true,
+	property = {"verify.process.name=com.liferay.file.uploads.configuration"},
+	service = VerifyProcess.class
+)
+public class GeneralFileUploadsVerifyProcess extends VerifyProcess {
```
Comment (JorgeFerrer) [inline modules/apps/foundation/file-uploads/file-uploads-configuration/src/main/java/com/liferay/file/uploads/configuration/verify/GeneralFileUploadsVerifyProcess.java]: Why is this a VerifyProcess and not an UpgradeProcess?

**Review by JorgeFerrer — CHANGES_REQUESTED** [review]
Hey Drew,

Nice job on this and thanks for sending it to me for review.

Overall, it looks good. I have a few questions, some of which I added as comments. But the main one I have is whether it's a good idea to put this in a new module. After all, this is just configuring a certain "service" that takes care of uploads, isn't it?
I tried to look for the code that is responsible for honoring this configuration. Do you know where is it? If it's in a module, we should consider putting the configuration along with it. If it's not, we should either consider extracting it or at least, think through where it would be extracted to put the configuration there.

File: modules/apps/foundation/file-uploads/file-uploads-configuration/src/main/java/com/liferay/file/uploads/configuration/verify/GeneralFileUploadsVerifyProcess.java (hunk 3 above)
Comment (drewbrokke) [inline modules/apps/foundation/file-uploads/file-uploads-configuration/src/main/java/com/liferay/file/uploads/configuration/verify/GeneralFileUploadsVerifyProcess.java]: I didn't see this kind of change being made in any of the Upgrade patterns, so I went with a VerifyProcess for this "custom pattern".  I'm running into the same issue with the Shopping File Uploads upgrade/verify - it seems like I'm using a new pattern.

Comment (JorgeFerrer) [inline modules/apps/foundation/file-uploads/file-uploads-configuration/src/main/java/com/liferay/file/uploads/configuration/verify/GeneralFileUploadsVerifyProcess.java]: Hey Drew,

You may want to check with Brian, since he's the one who usually creates the upgrade patterns. And I also think it was him to promoted using Verifiers a lot but lately realized or their drawbacks and suggests to avoid them when possible.
